<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Calculate phasograms (same strand alignment distances). — calcPhasogram • swissknife</title>


<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->
<link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/cerulean/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous" />


<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>




<meta property="og:title" content="Calculate phasograms (same strand alignment distances). — calcPhasogram" />
<meta property="og:description" content="Calculate the frequencies of same strand alignment distances,
  for example from MNase-seq data to estimate nucleosome repeat length.
  Distance calculations are implemented in C++ (calcAndCountDist)
  for efficiency." />






<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">swissknife</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.25</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/fmicompbio/swissknife/">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Calculate phasograms (same strand alignment distances).</h1>
    <small class="dont-index">Source: <a href='https://github.com/fmicompbio/swissknife/blob/master/R/phasograms.R'><code>R/phasograms.R</code></a></small>
    <div class="hidden name"><code>calcPhasogram.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Calculate the frequencies of same strand alignment distances,
  for example from MNase-seq data to estimate nucleosome repeat length.
  Distance calculations are implemented in C++ (<code><a href='calcAndCountDist.html'>calcAndCountDist</a></code>)
  for efficiency.</p>
    </div>

    <pre class="usage"><span class='fu'>calcPhasogram</span><span class='op'>(</span><span class='va'>fname</span>, regions <span class='op'>=</span> <span class='cn'>NULL</span>, rmdup <span class='op'>=</span> <span class='cn'>TRUE</span>, dmax <span class='op'>=</span> <span class='fl'>3000L</span><span class='op'>)</span></pre>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>fname</th>
      <td><p><code>character</code> vector with one or several bam files. If
multiple files are given, distance counts from all will be summed.</p></td>
    </tr>
    <tr>
      <th>regions</th>
      <td><p><code>GRanges</code> object. Only alignments falling into these
regions will be used. If <code>NULL</code> (the default), all alignments are
used.</p></td>
    </tr>
    <tr>
      <th>rmdup</th>
      <td><p><code><a href='https://rdrr.io/r/base/logical.html'>logical(1)</a></code> indicating if duplicates should be removed. If
<code>TRUE</code> (the default), only one of several alignments starting at the
same coordinate is used.</p></td>
    </tr>
    <tr>
      <th>dmax</th>
      <td><p><code><a href='https://rdrr.io/r/base/numeric.html'>numeric(1)</a></code> specifying the maximal distance between same
strand alignments to count.</p></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p><code>integer</code> vector with <code>dmax</code> elements, with the element at
  position <code>d</code> giving the observed number of alignment pairs at that
  distance.</p>
    <h2 class="hasAnchor" id="references"><a class="anchor" href="#references"></a>References</h2>

    <p>Phasograms were originally described in Valouev et al., Nature
  2011 (doi:10.1038/nature10002). The implementation here differs in two ways
  from the original algorithms:</p><ol>
<li><p>It does not implement removing of positions that have been seen less
    than <code>n</code> times (referred to as a <code>n</code>-pile subset in the paper).</p></li>
<li><p>It does allow to retain only alignments that fall into selected
    genomic intervals (<code>regions</code> argument).</p></li>
</ol>

    <h2 class="hasAnchor" id="see-also"><a class="anchor" href="#see-also"></a>See also</h2>

    <div class='dont-index'><p><code><a href='estimateNRL.html'>estimateNRL</a></code> to estimate the nucleosome repeat length
  from a phasogram, <code><a href='plotPhasogram.html'>plotPhasogram</a></code> to visualize an annotated
  phasogram, <code><a href='calcAndCountDist.html'>calcAndCountDist</a></code> for low-level distance counting.</p></div>
    <h2 class="hasAnchor" id="author"><a class="anchor" href="#author"></a>Author</h2>

    <p>Michael Stadler</p>

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'><span class='va'>bamf</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/system.file.html'>system.file</a></span><span class='op'>(</span><span class='st'>"extdata"</span>, <span class='st'>"phasograms"</span>, <span class='st'>"mnase_mm10.bam"</span>,
                    package <span class='op'>=</span> <span class='st'>"swissknife"</span><span class='op'>)</span>
<span class='va'>pg</span> <span class='op'>&lt;-</span> <span class='fu'>calcPhasogram</span><span class='op'>(</span><span class='va'>bamf</span><span class='op'>)</span>
<span class='fu'><a href='estimateNRL.html'>estimateNRL</a></span><span class='op'>(</span><span class='va'>pg</span>, usePeaks <span class='op'>=</span> <span class='fl'>1</span><span class='op'>:</span><span class='fl'>4</span><span class='op'>)</span><span class='op'>[</span><span class='fl'>1</span><span class='op'>:</span><span class='fl'>2</span><span class='op'>]</span>
</div><div class='output co'>#&gt; $nrl
#&gt; [1] 186
#&gt; 
#&gt; $nrl.CI95
#&gt;    2.5 %   97.5 % 
#&gt; 176.0015 195.9985 
#&gt; </div><div class='input'><span class='fu'><a href='plotPhasogram.html'>plotPhasogram</a></span><span class='op'>(</span><span class='va'>pg</span>, usePeaks <span class='op'>=</span> <span class='fl'>1</span><span class='op'>:</span><span class='fl'>4</span>, xlim <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='fl'>0</span>,<span class='fl'>1000</span><span class='op'>)</span><span class='op'>)</span>
</div><div class='img'><img src='calcPhasogram-1.png' alt='' width='700' height='433' /></div><div class='input'>
</div></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by Michael Stadler, <a href='https://csoneson.github.io'>Charlotte Soneson</a>, Panagiotis Papasaikas, Dania Machlab.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.9000.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


