<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Calculate phasograms (same strand alignment distances). — calcPhasogram • swissknife</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/cerulean/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Calculate phasograms (same strand alignment distances). — calcPhasogram"><meta property="og:description" content="Calculate the frequencies of same strand alignment distances,
  for example from MNase-seq data to estimate nucleosome repeat length.
  Distance calculations are implemented in C++ (calcAndCountDist)
  for efficiency."><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">swissknife</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.38</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/fmicompbio/swissknife/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Calculate phasograms (same strand alignment distances).</h1>
    <small class="dont-index">Source: <a href="https://github.com/fmicompbio/swissknife/blob/HEAD/R/phasograms.R" class="external-link"><code>R/phasograms.R</code></a></small>
    <div class="hidden name"><code>calcPhasogram.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Calculate the frequencies of same strand alignment distances,
  for example from MNase-seq data to estimate nucleosome repeat length.
  Distance calculations are implemented in C++ (<code><a href="calcAndCountDist.html">calcAndCountDist</a></code>)
  for efficiency.</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">calcPhasogram</span><span class="op">(</span><span class="va">fname</span>, regions <span class="op">=</span> <span class="cn">NULL</span>, rmdup <span class="op">=</span> <span class="cn">TRUE</span>, dmax <span class="op">=</span> <span class="fl">3000L</span><span class="op">)</span></span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>
    <dl><dt>fname</dt>
<dd><p><code>character</code> vector with one or several bam files. If
multiple files are given, distance counts from all will be summed.</p></dd>


<dt>regions</dt>
<dd><p><code>GRanges</code> object. Only alignments falling into these
regions will be used. If <code>NULL</code> (the default), all alignments are
used.</p></dd>


<dt>rmdup</dt>
<dd><p><code>logical(1)</code> indicating if duplicates should be removed. If
<code>TRUE</code> (the default), only one of several alignments starting at the
same coordinate is used.</p></dd>


<dt>dmax</dt>
<dd><p><code>numeric(1)</code> specifying the maximal distance between same
strand alignments to count.</p></dd>

</dl></div>
    <div id="value">
    <h2>Value</h2>
    

<p><code>integer</code> vector with <code>dmax</code> elements, with the element at
  position <code>d</code> giving the observed number of alignment pairs at that
  distance.</p>
    </div>
    <div id="references">
    <h2>References</h2>
    <p>Phasograms were originally described in Valouev et al., Nature
  2011 (doi:10.1038/nature10002). The implementation here differs in two ways
  from the original algorithms:</p><ol><li><p>It does not implement removing of positions that have been seen less
    than <code>n</code> times (referred to as a <code>n</code>-pile subset in the paper).</p></li>
<li><p>It does allow to retain only alignments that fall into selected
    genomic intervals (<code>regions</code> argument).</p></li>
</ol></div>
    <div id="see-also">
    <h2>See also</h2>
    <div class="dont-index"><p><code><a href="estimateNRL.html">estimateNRL</a></code> to estimate the nucleosome repeat length
  from a phasogram, <code><a href="plotPhasogram.html">plotPhasogram</a></code> to visualize an annotated
  phasogram, <code><a href="calcAndCountDist.html">calcAndCountDist</a></code> for low-level distance counting.</p></div>
    </div>
    <div id="author">
    <h2>Author</h2>
    <p>Michael Stadler</p>
    </div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="kw">if</span> <span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html" class="external-link">requireNamespace</a></span><span class="op">(</span><span class="st">"GenomicAlignments"</span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span> <span class="op">&amp;&amp;</span></span></span>
<span class="r-in"><span>    <span class="fu"><a href="https://rdrr.io/r/base/ns-load.html" class="external-link">requireNamespace</a></span><span class="op">(</span><span class="st">"Rsamtools"</span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span></span>
<span class="r-in"><span>    <span class="va">bamf</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"phasograms"</span>, <span class="st">"mnase_mm10.bam"</span>,</span></span>
<span class="r-in"><span>                        package <span class="op">=</span> <span class="st">"swissknife"</span><span class="op">)</span></span></span>
<span class="r-in"><span>    <span class="va">pg</span> <span class="op">&lt;-</span> <span class="fu">calcPhasogram</span><span class="op">(</span><span class="va">bamf</span><span class="op">)</span></span></span>
<span class="r-in"><span>    <span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="fu"><a href="estimateNRL.html">estimateNRL</a></span><span class="op">(</span><span class="va">pg</span>, usePeaks <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">4</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">2</span><span class="op">]</span><span class="op">)</span></span></span>
<span class="r-in"><span>    <span class="fu"><a href="plotPhasogram.html">plotPhasogram</a></span><span class="op">(</span><span class="va">pg</span>, usePeaks <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">4</span>, xlim <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">1000</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="op">}</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> $nrl</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] 186</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> $nrl.CI95</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    2.5 %   97.5 % </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 176.0015 195.9985 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-plt img"><img src="calcPhasogram-1.png" alt="" width="700" height="433"></span>
<span class="r-in"><span></span></span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Michael Stadler, <a href="https://csoneson.github.io" class="external-link">Charlotte Soneson</a>, Panagiotis Papasaikas, Dania Machlab.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.6.9000.</p>
</div>

      </footer></div>

  


  

  </body></html>

