<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Get regions tiling a genome. — getGenomicTiles • swissknife</title>


<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->
<link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/cerulean/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous" />


<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>




<meta property="og:title" content="Get regions tiling a genome. — getGenomicTiles" />
<meta property="og:description" content="Get sequential, potentially annotated regions of a fixed lengths
   (tiles) along chromosomes of a genome." />






<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



<!-- dependencies from examples -->


  

  

  </head>

  <body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">swissknife</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.30</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/fmicompbio/swissknife/">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Get regions tiling a genome.</h1>
    <small class="dont-index">Source: <a href='https://github.com/fmicompbio/swissknife/blob/master/R/getGenomicTiles.R'><code>R/getGenomicTiles.R</code></a></small>
    <div class="hidden name"><code>getGenomicTiles.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Get sequential, potentially annotated regions of a fixed lengths
   (tiles) along chromosomes of a genome.</p>
    </div>

    <pre class="usage"><span class='fu'>getGenomicTiles</span><span class='op'>(</span>
  <span class='va'>genome</span>,
  <span class='va'>tileWidth</span>,
  hasOverlap <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/list.html'>list</a></span><span class='op'>(</span><span class='op'>)</span>,
  fracOverlap <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/list.html'>list</a></span><span class='op'>(</span><span class='op'>)</span>,
  numOverlap <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/list.html'>list</a></span><span class='op'>(</span><span class='op'>)</span>,
  nearest <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/list.html'>list</a></span><span class='op'>(</span><span class='op'>)</span>,
  addSeqComp <span class='op'>=</span> <span class='cn'>TRUE</span>
<span class='op'>)</span></pre>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>genome</th>
      <td><p>The genome to work on. Either a <code><a href='https://rdrr.io/pkg/BSgenome/man/BSgenome-class.html'>BSgenome</a></code>
object, a <code>character</code> scalar with the name of an installed <code><a href='https://rdrr.io/pkg/BSgenome/man/BSgenome-class.html'>BSgenome</a></code>
or with a file path and name pointing to a fasta file with the genome sequence,
or a named <code>numeric</code> vector giving the names and lengths of chromosomes.</p></td>
    </tr>
    <tr>
      <th>tileWidth</th>
      <td><p><code>numeric</code> scalar with the tile length.</p></td>
    </tr>
    <tr>
      <th>hasOverlap, fracOverlap, numOverlap, nearest</th>
      <td><p>Named <code>list</code>s with
<code><a href='https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html'>GRanges</a></code> or <code><a href='https://rdrr.io/pkg/GenomicRanges/man/GRangesList-class.html'>GRangesList</a></code>
object(s) used to annotate genomic tiles. See <code><a href='annotateRegions.html'>annotateRegions</a></code>
for details.</p></td>
    </tr>
    <tr>
      <th>addSeqComp</th>
      <td><p><code>logical</code> scalar. If <code>TRUE</code> and primary sequence
can be obtained from <code>genome</code>, also add sequence composition features
for each tile to the annotations. Currently, the following features are
included: percent of G+C bases ("percGC"), CpG observed-over-expected ratio
("CpGoe").</p></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>A <code><a href='https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html'>GRanges</a></code> object with genome tiling regions.
    Optional tile annotations are contained in its metadata columns (<code>mcols</code>).</p>
    <h2 class="hasAnchor" id="details"><a class="anchor" href="#details"></a>Details</h2>

    <p>The last tile in each chromosome is dropped if it would be shorter
    than <code>tileWidth</code>. Generated tiles are unstranded (<code><a href='https://rdrr.io/r/base/Arithmetic.html'>*</a></code>) and
    therefore overlaps or searching for nearest neighbors are ignoring
    strands of annotations (<code>ignore.strand=TRUE</code>).</p>
    <h2 class="hasAnchor" id="see-also"><a class="anchor" href="#see-also"></a>See also</h2>

    <div class='dont-index'><p><code><a href='https://rdrr.io/pkg/GenomicRanges/man/tileGenome.html'>tileGenome</a></code> and <code><a href='annotateRegions.html'>annotateRegions</a></code>
    used by <code>getGenomicTiles</code> internally.</p></div>
    <h2 class="hasAnchor" id="author"><a class="anchor" href="#author"></a>Author</h2>

    <p>Michael Stadler</p>

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'><span class='kw'><a href='https://rdrr.io/r/base/library.html'>library</a></span><span class='op'>(</span><span class='va'><a href='https://bioconductor.org/packages/GenomicRanges'>GenomicRanges</a></span><span class='op'>)</span>

<span class='va'>tss</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html'>GRanges</a></span><span class='op'>(</span><span class='st'>"chr1"</span>, <span class='fu'>IRanges</span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='fl'>1</span>, <span class='fl'>10</span>, <span class='fl'>30</span><span class='op'>)</span>, width <span class='op'>=</span> <span class='fl'>1</span>,
                               names <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/paste.html'>paste0</a></span><span class='op'>(</span><span class='st'>"t"</span>, <span class='fl'>1</span><span class='op'>:</span><span class='fl'>3</span><span class='op'>)</span><span class='op'>)</span><span class='op'>)</span>
<span class='va'>blacklist</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html'>GRanges</a></span><span class='op'>(</span><span class='st'>"chr1"</span>, <span class='fu'>IRanges</span><span class='op'>(</span><span class='fl'>20</span>, width <span class='op'>=</span> <span class='fl'>5</span><span class='op'>)</span><span class='op'>)</span>
<span class='fu'>getGenomicTiles</span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span>chr1 <span class='op'>=</span> <span class='fl'>45</span>, chr2 <span class='op'>=</span> <span class='fl'>12</span><span class='op'>)</span>, tileWidth <span class='op'>=</span> <span class='fl'>10</span>,
                hasOverlap <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/list.html'>list</a></span><span class='op'>(</span>Blacklist <span class='op'>=</span> <span class='va'>blacklist</span><span class='op'>)</span>,
                fracOverlap <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/list.html'>list</a></span><span class='op'>(</span>Blacklist <span class='op'>=</span> <span class='va'>blacklist</span><span class='op'>)</span>,
                numOverlap <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/list.html'>list</a></span><span class='op'>(</span>TSS <span class='op'>=</span> <span class='va'>tss</span><span class='op'>)</span>,
                nearest <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/list.html'>list</a></span><span class='op'>(</span>TSS <span class='op'>=</span> <span class='va'>tss</span><span class='op'>)</span><span class='op'>)</span>
</div><div class='output co'>#&gt; <span class='warning'>Warning: ignoring 'addSeqComp' (sequence not available from 'genome')</span></div><div class='output co'>#&gt; GRanges object with 5 ranges and 6 metadata columns:
#&gt;       seqnames    ranges strand | Blacklist.hasOverlap Blacklist.fracOverlap
#&gt;          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; |            &lt;logical&gt;             &lt;numeric&gt;
#&gt;   [1]     chr1      1-10      * |                FALSE                   0.0
#&gt;   [2]     chr1     11-20      * |                 TRUE                   0.1
#&gt;   [3]     chr1     21-30      * |                 TRUE                   0.4
#&gt;   [4]     chr1     31-40      * |                FALSE                   0.0
#&gt;   [5]     chr2      1-10      * |                FALSE                   0.0
#&gt;       TSS.numOverlapWithin TSS.numOverlapAny TSS.nearestName
#&gt;                  &lt;integer&gt;         &lt;integer&gt;     &lt;character&gt;
#&gt;   [1]                    2                 2              t1
#&gt;   [2]                    0                 0              t2
#&gt;   [3]                    1                 1              t3
#&gt;   [4]                    0                 0              t3
#&gt;   [5]                    0                 0            &lt;NA&gt;
#&gt;       TSS.nearestDistance
#&gt;                 &lt;integer&gt;
#&gt;   [1]                   0
#&gt;   [2]                   0
#&gt;   [3]                   0
#&gt;   [4]                   0
#&gt;   [5]                &lt;NA&gt;
#&gt;   -------
#&gt;   seqinfo: 2 sequences from an unspecified genome</div><div class='input'>
</div></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p><p>Developed by Michael Stadler, <a href="https://csoneson.github.io" class="external-link">Charlotte Soneson</a>, Panagiotis Papasaikas, Dania Machlab.</p></p>
</div>

<div class="pkgdown">
  <p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 1.6.1.9001.</p></p>
</div>

      </footer>
   </div>

  


  

  </body>
</html>


