<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Get regions tiling a genome. — getGenomicTiles • swissknife</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/cerulean/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Get regions tiling a genome. — getGenomicTiles"><meta property="og:description" content="Get sequential, potentially annotated regions of a fixed lengths
   (tiles) along chromosomes of a genome."><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">swissknife</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.35</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/fmicompbio/swissknife/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Get regions tiling a genome.</h1>
    <small class="dont-index">Source: <a href="https://github.com/fmicompbio/swissknife/blob/HEAD/R/getGenomicTiles.R" class="external-link"><code>R/getGenomicTiles.R</code></a></small>
    <div class="hidden name"><code>getGenomicTiles.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Get sequential, potentially annotated regions of a fixed lengths
   (tiles) along chromosomes of a genome.</p>
    </div>

    <div class="ref-usage sourceCode"><pre class="sourceCode r"><code><span class="fu">getGenomicTiles</span><span class="op">(</span>
  <span class="va">genome</span>,
  <span class="va">tileWidth</span>,
  hasOverlap <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="op">)</span>,
  fracOverlap <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="op">)</span>,
  numOverlap <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="op">)</span>,
  nearest <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="op">)</span>,
  addSeqComp <span class="op">=</span> <span class="cn">TRUE</span>
<span class="op">)</span></code></pre></div>

    <div id="ref-sections">
    <div id="arguments">
    <h2>Arguments</h2>
    <dl><dt>genome</dt>
<dd><p>The genome to work on. Either a <code><a href="https://rdrr.io/pkg/BSgenome/man/BSgenome-class.html" class="external-link">BSgenome</a></code>
object, a <code>character</code> scalar with the name of an installed <code><a href="https://rdrr.io/pkg/BSgenome/man/BSgenome-class.html" class="external-link">BSgenome</a></code>
or with a file path and name pointing to a fasta file with the genome sequence,
or a named <code>numeric</code> vector giving the names and lengths of chromosomes.</p></dd>
<dt>tileWidth</dt>
<dd><p><code>numeric</code> scalar with the tile length.</p></dd>
<dt>hasOverlap, fracOverlap, numOverlap, nearest</dt>
<dd><p>Named <code>list</code>s with
<code><a href="https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html" class="external-link">GRanges</a></code> or <code><a href="https://rdrr.io/pkg/GenomicRanges/man/GRangesList-class.html" class="external-link">GRangesList</a></code>
object(s) used to annotate genomic tiles. See <code><a href="annotateRegions.html">annotateRegions</a></code>
for details.</p></dd>
<dt>addSeqComp</dt>
<dd><p><code>logical</code> scalar. If <code>TRUE</code> and primary sequence
can be obtained from <code>genome</code>, also add sequence composition features
for each tile to the annotations. Currently, the following features are
included: percent of G+C bases ("percGC"), CpG observed-over-expected ratio
("CpGoe").</p></dd>
</dl></div>
    <div id="value">
    <h2>Value</h2>
    <p>A <code><a href="https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html" class="external-link">GRanges</a></code> object with genome tiling regions.
    Optional tile annotations are contained in its metadata columns (<code>mcols</code>).</p>
    </div>
    <div id="details">
    <h2>Details</h2>
    <p>The last tile in each chromosome is dropped if it would be shorter
    than <code>tileWidth</code>. Generated tiles are unstranded (<code>*</code>) and
    therefore overlaps or searching for nearest neighbors are ignoring
    strands of annotations (<code>ignore.strand=TRUE</code>).</p>
    </div>
    <div id="see-also">
    <h2>See also</h2>
    <div class="dont-index"><p><code><a href="https://rdrr.io/pkg/GenomicRanges/man/tileGenome.html" class="external-link">tileGenome</a></code> and <code><a href="annotateRegions.html">annotateRegions</a></code>
    used by <code>getGenomicTiles</code> internally.</p></div>
    </div>
    <div id="author">
    <h2>Author</h2>
    <p>Michael Stadler</p>
    </div>
    </div>

    <div id="examples">
    <h2>Examples</h2>
    <div class="ref-examples sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://bioconductor.org/packages/GenomicRanges" class="external-link">GenomicRanges</a></span><span class="op">)</span></span>
<span class="r-in"></span>
<span class="r-in"><span class="va">tss</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html" class="external-link">GRanges</a></span><span class="op">(</span><span class="st">"chr1"</span>, <span class="fu"><a href="https://rdrr.io/pkg/IRanges/man/IRanges-constructor.html" class="external-link">IRanges</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">10</span>, <span class="fl">30</span><span class="op">)</span>, width <span class="op">=</span> <span class="fl">1</span>,</span>
<span class="r-in">                               names <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"t"</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">3</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span class="r-in"><span class="va">blacklist</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html" class="external-link">GRanges</a></span><span class="op">(</span><span class="st">"chr1"</span>, <span class="fu"><a href="https://rdrr.io/pkg/IRanges/man/IRanges-constructor.html" class="external-link">IRanges</a></span><span class="op">(</span><span class="fl">20</span>, width <span class="op">=</span> <span class="fl">5</span><span class="op">)</span><span class="op">)</span></span>
<span class="r-in"><span class="fu">getGenomicTiles</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">c</a></span><span class="op">(</span>chr1 <span class="op">=</span> <span class="fl">45</span>, chr2 <span class="op">=</span> <span class="fl">12</span><span class="op">)</span>, tileWidth <span class="op">=</span> <span class="fl">10</span>,</span>
<span class="r-in">                hasOverlap <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>Blacklist <span class="op">=</span> <span class="va">blacklist</span><span class="op">)</span>,</span>
<span class="r-in">                fracOverlap <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>Blacklist <span class="op">=</span> <span class="va">blacklist</span><span class="op">)</span>,</span>
<span class="r-in">                numOverlap <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>TSS <span class="op">=</span> <span class="va">tss</span><span class="op">)</span>,</span>
<span class="r-in">                nearest <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>TSS <span class="op">=</span> <span class="va">tss</span><span class="op">)</span><span class="op">)</span></span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>ignoring 'addSeqComp' (sequence not available from 'genome')</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GRanges object with 5 ranges and 6 metadata columns:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       seqnames    ranges strand | Blacklist.hasOverlap Blacklist.fracOverlap</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; |            &lt;logical&gt;             &lt;numeric&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [1]     chr1      1-10      * |                FALSE                   0.0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [2]     chr1     11-20      * |                 TRUE                   0.1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [3]     chr1     21-30      * |                 TRUE                   0.4</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [4]     chr1     31-40      * |                FALSE                   0.0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [5]     chr2      1-10      * |                FALSE                   0.0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       TSS.numOverlapWithin TSS.numOverlapAny TSS.nearestName</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                  &lt;integer&gt;         &lt;integer&gt;     &lt;character&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [1]                    2                 2              t1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [2]                    0                 0              t2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [3]                    1                 1              t3</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [4]                    0                 0              t3</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [5]                    0                 0            &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       TSS.nearestDistance</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                 &lt;integer&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [1]                   0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [2]                   0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [3]                   0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [4]                   0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [5]                &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   -------</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   seqinfo: 2 sequences from an unspecified genome</span>
<span class="r-in"></span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Michael Stadler, <a href="https://csoneson.github.io" class="external-link">Charlotte Soneson</a>, Panagiotis Papasaikas, Dania Machlab.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 1.9000.9000.9000.</p>
</div>

      </footer></div>

  


  

  </body></html>

