<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Get mappable regions of a genome. — getMappableRegions • swissknife</title>


<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->
<link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/cerulean/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous" />


<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>




<meta property="og:title" content="Get mappable regions of a genome. — getMappableRegions" />
<meta property="og:description" content="Given a k-mer length and the maximum number of allowed hits per
    k-mer, find all mappable regions in a genome." />






<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



<!-- dependencies from examples -->


  

  

  </head>

  <body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">swissknife</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.30</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/fmicompbio/swissknife/">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Get mappable regions of a genome.</h1>
    <small class="dont-index">Source: <a href='https://github.com/fmicompbio/swissknife/blob/master/R/getMappableRegions.R'><code>R/getMappableRegions.R</code></a></small>
    <div class="hidden name"><code>getMappableRegions.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Given a k-mer length and the maximum number of allowed hits per
    k-mer, find all mappable regions in a genome.</p>
    </div>

    <pre class="usage"><span class='fu'>getMappableRegions</span><span class='op'>(</span>
  <span class='va'>genome</span>,
  <span class='va'>genomeIndex</span>,
  kmerLength <span class='op'>=</span> <span class='fl'>50</span>,
  maxHits <span class='op'>=</span> <span class='fl'>1</span>,
  Ncpu <span class='op'>=</span> <span class='fl'>2</span>,
  quiet <span class='op'>=</span> <span class='cn'>TRUE</span>
<span class='op'>)</span></pre>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>genome</th>
      <td><p>The genome sequence to work on. Either a <code><a href='https://rdrr.io/pkg/BSgenome/man/BSgenome-class.html'>BSgenome</a></code>
object, a <code>character</code> scalar with the name of an installed <code><a href='https://rdrr.io/pkg/BSgenome/man/BSgenome-class.html'>BSgenome</a></code>
or with a file path and name pointing to a fasta file with the genome sequence.</p></td>
    </tr>
    <tr>
      <th>genomeIndex</th>
      <td><p><code>character</code> scalar with the path to the bowtie index
and prefix to align against, in the form <code>&lt;/path/to/index&gt;/&lt;prefix&gt;</code>,
or the name of an installed <code>Rbowtie</code> index package created by the
<span class="pkg">QuasR</span> package for an installed <code>BSgenome</code> package.</p></td>
    </tr>
    <tr>
      <th>kmerLength</th>
      <td><p><code>numeric</code> scalar specifying the k-mer length (width of
overlapping windows in <code>genome</code>), usually set to the typical read
length for which to get the mappable regions.</p></td>
    </tr>
    <tr>
      <th>maxHits</th>
      <td><p><code>numeric</code> scalar specifying the maximum number of hits
(matches) of a k-mer in the <code>genome</code> to be considered mappable.</p></td>
    </tr>
    <tr>
      <th>Ncpu</th>
      <td><p><code>numeric</code> scalar specifying the number of CPU threads to use
for alignments.</p></td>
    </tr>
    <tr>
      <th>quiet</th>
      <td><p><code>logical</code> scalar indicating if progress information should
be printed on the console.</p></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>A <code><a href='https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html'>GRanges</a></code> object with mappable regions.
    All plus-strand sequences in <code>genome</code> of length <code>kmerLength</code>
    with their start (leftmost) position overlapping the <code>GRanges</code> object
    do not generate more than <code>maxHits</code> hits when aligned to the genome.</p>
    <h2 class="hasAnchor" id="details"><a class="anchor" href="#details"></a>Details</h2>

    <p>Sequences of all overlapping windows are extracted from the genome
    and aligned to the provided genome index using <code><a href='https://rdrr.io/pkg/Rbowtie/man/Rbowtie.html'>bowtie</a></code>
    with parameters <code>-f -v 0 -a -B 1 -m maxHits</code>. If no more
    than <code>maxHits</code> hits are found, the window is defined mappable.</p>
    <h2 class="hasAnchor" id="see-also"><a class="anchor" href="#see-also"></a>See also</h2>

    <div class='dont-index'><p><code><a href='https://rdrr.io/pkg/Rbowtie/man/Rbowtie.html'>bowtie</a></code> in package <span class="pkg">Rbowtie</span> used by
    <code>getMappableRegions</code> to align reads to the genome;
    <code><a href='https://rdrr.io/pkg/Rbowtie/man/Rbowtie.html'>bowtie_build</a></code> in package <span class="pkg">Rbowtie</span> for
    indexing a genome.</p></div>
    <h2 class="hasAnchor" id="author"><a class="anchor" href="#author"></a>Author</h2>

    <p>Michael Stadler</p>

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'><span class='kw'>if</span> <span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/ns-load.html'>requireNamespace</a></span><span class='op'>(</span><span class='st'>"Rbowtie"</span>, quietly <span class='op'>=</span> <span class='cn'>TRUE</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>{</span>
    <span class='kw'><a href='https://rdrr.io/r/base/library.html'>library</a></span><span class='op'>(</span><span class='va'>Rbowtie</span><span class='op'>)</span>

    <span class='va'>genomefile</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/system.file.html'>system.file</a></span><span class='op'>(</span><span class='st'>"extdata"</span>, <span class='st'>"getMappableRegions"</span>, <span class='st'>"hg19sub.fa"</span>, package <span class='op'>=</span> <span class='st'>"swissknife"</span><span class='op'>)</span>
    <span class='va'>indexdir</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/tempfile.html'>tempfile</a></span><span class='op'>(</span><span class='op'>)</span>
    <span class='va'>indexpre</span> <span class='op'>&lt;-</span> <span class='st'>"index"</span>
    <span class='va'>indexname</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/file.path.html'>file.path</a></span><span class='op'>(</span><span class='va'>indexdir</span>, <span class='va'>indexpre</span><span class='op'>)</span>
    <span class='va'>idx</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/Rbowtie/man/Rbowtie.html'>bowtie_build</a></span><span class='op'>(</span><span class='va'>genomefile</span>, <span class='va'>indexdir</span><span class='op'>)</span>

    <span class='va'>mapgr</span> <span class='op'>&lt;-</span> <span class='fu'>getMappableRegions</span><span class='op'>(</span><span class='va'>genomefile</span>, <span class='va'>indexname</span>, <span class='fl'>50</span>, quiet <span class='op'>=</span> <span class='cn'>FALSE</span><span class='op'>)</span>
    <span class='fu'><a href='https://rdrr.io/r/base/print.html'>print</a></span><span class='op'>(</span><span class='va'>mapgr</span><span class='op'>)</span>
<span class='op'>}</span>
</div><div class='output co'>#&gt; <span class='message'>Will calculate mappability for 3 chromosomes (95000 base pairs)</span></div><div class='output co'>#&gt; <span class='message'>    chr1...</span></div><div class='output co'>#&gt; <span class='message'>done (2.1% non-mappable)</span></div><div class='output co'>#&gt; <span class='message'>    chr2...</span></div><div class='output co'>#&gt; <span class='message'>done (0.2% non-mappable)</span></div><div class='output co'>#&gt; <span class='message'>    chr3...</span></div><div class='output co'>#&gt; <span class='message'>done (0.1% non-mappable)</span></div><div class='output co'>#&gt; <span class='message'>Total mappability: 99.2% in 27 segments</span></div><div class='output co'>#&gt; GRanges object with 27 ranges and 0 metadata columns:
#&gt;        seqnames      ranges strand
#&gt;           &lt;Rle&gt;   &lt;IRanges&gt;  &lt;Rle&gt;
#&gt;    [1]     chr1      1-2158      +
#&gt;    [2]     chr1   2619-2649      +
#&gt;    [3]     chr1   2657-2666      +
#&gt;    [4]     chr1   2693-2704      +
#&gt;    [5]     chr1   2788-2818      +
#&gt;    ...      ...         ...    ...
#&gt;   [23]     chr3 23773-31758      +
#&gt;   [24]     chr3 31761-33858      +
#&gt;   [25]     chr3 33867-35084      +
#&gt;   [26]     chr3 35099-44383      +
#&gt;   [27]     chr3 44387-45000      +
#&gt;   -------
#&gt;   seqinfo: 3 sequences from an unspecified genome</div><div class='input'>
</div></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p><p>Developed by Michael Stadler, <a href="https://csoneson.github.io" class="external-link">Charlotte Soneson</a>, Panagiotis Papasaikas, Dania Machlab.</p></p>
</div>

<div class="pkgdown">
  <p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 1.6.1.9001.</p></p>
</div>

      </footer>
   </div>

  


  

  </body>
</html>


