<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Assign labels to cells using known marker genes — labelCells • swissknife</title>


<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->
<link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/cerulean/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous" />


<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>




<meta property="og:title" content="Assign labels to cells using known marker genes — labelCells" />
<meta property="og:description" content="Given marker gene sets for cell types, identify cells with
  high expression of the marker genes (positive examples), then use these
  cells to create a reference transcriptome profile for each cell type and
  identify additional cells of each type using SingleR. These
  marker genes should specifically expressed a single cell type, e.g.
  CD3 which is expressed by all T cell subtypes would not be suitable
  for specific T cell subtypes." />






<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



<!-- dependencies from examples -->


  

  

  </head>

  <body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">swissknife</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.31</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/fmicompbio/swissknife/">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Assign labels to cells using known marker genes</h1>
    <small class="dont-index">Source: <a href='https://github.com/fmicompbio/swissknife/blob/master/R/cellLabeling.R'><code>R/cellLabeling.R</code></a></small>
    <div class="hidden name"><code>labelCells.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Given marker gene sets for cell types, identify cells with
  high expression of the marker genes (positive examples), then use these
  cells to create a reference transcriptome profile for each cell type and
  identify additional cells of each type using <code>SingleR</code>. These
  marker genes should specifically expressed a single cell type, e.g.
  CD3 which is expressed by all T cell subtypes would not be suitable
  for specific T cell subtypes.</p>
    </div>

    <pre class="usage"><span class='fu'>labelCells</span><span class='op'>(</span>
  <span class='va'>sce</span>,
  <span class='va'>markergenes</span>,
  fraction_topscoring <span class='op'>=</span> <span class='fl'>0.01</span>,
  expr_values <span class='op'>=</span> <span class='st'>"logcounts"</span>,
  normGenesetExpressionParams <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/list.html'>list</a></span><span class='op'>(</span>R <span class='op'>=</span> <span class='fl'>200</span><span class='op'>)</span>,
  aggregateReferenceParams <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/list.html'>list</a></span><span class='op'>(</span>power <span class='op'>=</span> <span class='fl'>0.5</span><span class='op'>)</span>,
  SingleRParams <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/list.html'>list</a></span><span class='op'>(</span><span class='op'>)</span>,
  BPPARAM <span class='op'>=</span> <span class='fu'>SerialParam</span><span class='op'>(</span><span class='op'>)</span>
<span class='op'>)</span></pre>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>sce</th>
      <td><p><code>SingleCellExperiment</code> object.</p></td>
    </tr>
    <tr>
      <th>markergenes</th>
      <td><p>Named <code>list</code> of <code>character</code> vectors with the
marker genes for each cell types. The marker genes must be a subset of
<code>rownames(sce)</code>.</p></td>
    </tr>
    <tr>
      <th>fraction_topscoring</th>
      <td><p><code>numeric</code> vector of length 1 or the same
length as <code>markergenes</code> giving the fraction(s) of top scoring cells
for each cell type to pick to create the reference transcriptome profile.</p></td>
    </tr>
    <tr>
      <th>expr_values</th>
      <td><p>Integer scalar or string indicating which assay of
<code>sce</code> contains the expression values.</p></td>
    </tr>
    <tr>
      <th>normGenesetExpressionParams</th>
      <td><p><code>list</code> with additional parameters
for <code><a href='normGenesetExpression.html'>normGenesetExpression</a></code>.</p></td>
    </tr>
    <tr>
      <th>aggregateReferenceParams</th>
      <td><p><code>list</code> with additional parameters
for <code><a href='https://rdrr.io/pkg/SingleR/man/aggregateReference.html'>aggregateReference</a></code>.</p></td>
    </tr>
    <tr>
      <th>SingleRParams</th>
      <td><p><code>list</code> with additional parameters for
<code><a href='https://rdrr.io/pkg/SingleR/man/SingleR.html'>SingleR</a></code>.</p></td>
    </tr>
    <tr>
      <th>BPPARAM</th>
      <td><p>An optional <code><a href='https://rdrr.io/pkg/BiocParallel/man/BiocParallelParam-class.html'>BiocParallelParam</a></code>
instance determining the parallel back-end to be used during evaluation.</p></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>A <code>list</code> of three elements named <code>cells</code>, <code>refs</code> and <code>labels</code>.
  <code>cells</code> contains a <code>list</code> with the numerical indices of the top
  scoring cells for each cell type.
  <code>refs</code> contains the pseudo-bulk transcriptome profiles used as a 
  reference for label assignment, as returned by <code><a href='https://rdrr.io/pkg/SingleR/man/aggregateReference.html'>aggregateReference</a></code>.
  <code>labels</code> contains a <code><a href='https://rdrr.io/pkg/S4Vectors/man/DataFrame-class.html'>DataFrame</a></code> with the
  annotation statistics for each cell (one cell per row), generated by
  <code><a href='https://rdrr.io/pkg/SingleR/man/SingleR.html'>SingleR</a></code>.</p>
    <h2 class="hasAnchor" id="see-also"><a class="anchor" href="#see-also"></a>See also</h2>

    <div class='dont-index'><p><code><a href='normGenesetExpression.html'>normGenesetExpression</a></code> used to calculate scores for
   marker gene sets; <code><a href='https://rdrr.io/pkg/SingleR/man/aggregateReference.html'>aggregateReference</a></code> used to
   create reference profiles; <code><a href='https://rdrr.io/pkg/SingleR/man/SingleR.html'>SingleR</a></code> used to assign
   labels to cells.</p></div>
    <h2 class="hasAnchor" id="author"><a class="anchor" href="#author"></a>Author</h2>

    <p>Michael Stadler</p>

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'><span class='kw'>if</span> <span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/ns-load.html'>requireNamespace</a></span><span class='op'>(</span><span class='st'>"SingleR"</span>, quietly <span class='op'>=</span> <span class='cn'>TRUE</span><span class='op'>)</span> <span class='op'>&amp;&amp;</span>
    <span class='fu'><a href='https://rdrr.io/r/base/ns-load.html'>requireNamespace</a></span><span class='op'>(</span><span class='st'>"SingleCellExperiment"</span>, quietly <span class='op'>=</span> <span class='cn'>TRUE</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>{</span>
    
    <span class='co'># create SingleCellExperiment with cell-type specific genes</span>
    <span class='kw'><a href='https://rdrr.io/r/base/library.html'>library</a></span><span class='op'>(</span><span class='va'>SingleCellExperiment</span><span class='op'>)</span>
    <span class='va'>n_types</span> <span class='op'>&lt;-</span> <span class='fl'>3</span>
    <span class='va'>n_per_type</span> <span class='op'>&lt;-</span> <span class='fl'>30</span>
    <span class='va'>n_cells</span> <span class='op'>&lt;-</span> <span class='va'>n_types</span> <span class='op'>*</span> <span class='va'>n_per_type</span>
    <span class='va'>n_genes</span> <span class='op'>&lt;-</span> <span class='fl'>500</span>
    <span class='va'>fraction_specific</span> <span class='op'>&lt;-</span> <span class='fl'>0.1</span>
    <span class='va'>n_specific</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/Round.html'>round</a></span><span class='op'>(</span><span class='va'>n_genes</span> <span class='op'>*</span> <span class='va'>fraction_specific</span><span class='op'>)</span>

    <span class='fu'><a href='https://rdrr.io/r/base/Random.html'>set.seed</a></span><span class='op'>(</span><span class='fl'>42</span><span class='op'>)</span>
    <span class='va'>mu</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/Round.html'>ceiling</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/stats/Uniform.html'>runif</a></span><span class='op'>(</span>n <span class='op'>=</span> <span class='va'>n_genes</span>, min <span class='op'>=</span> <span class='fl'>0</span>, max <span class='op'>=</span> <span class='fl'>30</span><span class='op'>)</span><span class='op'>)</span>
    <span class='va'>u</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/BiocGenerics/man/do.call.html'>do.call</a></span><span class='op'>(</span><span class='va'>rbind</span>, <span class='fu'><a href='https://rdrr.io/pkg/BiocGenerics/man/lapply.html'>lapply</a></span><span class='op'>(</span><span class='va'>mu</span>, <span class='kw'>function</span><span class='op'>(</span><span class='va'>x</span><span class='op'>)</span> <span class='fu'><a href='https://rdrr.io/r/stats/Poisson.html'>rpois</a></span><span class='op'>(</span><span class='va'>n_cells</span>, lambda <span class='op'>=</span> <span class='va'>x</span><span class='op'>)</span><span class='op'>)</span><span class='op'>)</span>
    <span class='fu'><a href='https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html'>rownames</a></span><span class='op'>(</span><span class='va'>u</span><span class='op'>)</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/paste.html'>paste0</a></span><span class='op'>(</span><span class='st'>"g"</span>, <span class='fu'><a href='https://rdrr.io/r/base/seq.html'>seq.int</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/pkg/BiocGenerics/man/nrow.html'>nrow</a></span><span class='op'>(</span><span class='va'>u</span><span class='op'>)</span><span class='op'>)</span><span class='op'>)</span>
    <span class='va'>celltype.labels</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/S4Vectors/man/Vector-class.html'>rep</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/paste.html'>paste0</a></span><span class='op'>(</span><span class='st'>"t"</span>, <span class='fu'><a href='https://rdrr.io/r/base/seq.html'>seq.int</a></span><span class='op'>(</span><span class='va'>n_types</span><span class='op'>)</span><span class='op'>)</span>, each <span class='op'>=</span> <span class='va'>n_per_type</span><span class='op'>)</span>
    <span class='va'>celltype.genes</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/S4Vectors/man/splitAsList.html'>split</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/sample.html'>sample</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html'>rownames</a></span><span class='op'>(</span><span class='va'>u</span><span class='op'>)</span>, size <span class='op'>=</span> <span class='va'>n_types</span> <span class='op'>*</span> <span class='va'>n_specific</span><span class='op'>)</span>,
                            <span class='fu'><a href='https://rdrr.io/pkg/S4Vectors/man/Vector-class.html'>rep</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/paste.html'>paste0</a></span><span class='op'>(</span><span class='st'>"t"</span>, <span class='fu'><a href='https://rdrr.io/r/base/seq.html'>seq.int</a></span><span class='op'>(</span><span class='va'>n_types</span><span class='op'>)</span><span class='op'>)</span>, each <span class='op'>=</span> <span class='va'>n_specific</span><span class='op'>)</span><span class='op'>)</span>
    <span class='kw'>for</span> <span class='op'>(</span><span class='va'>i</span> <span class='kw'>in</span> <span class='fu'><a href='https://rdrr.io/r/base/seq.html'>seq_along</a></span><span class='op'>(</span><span class='va'>celltype.genes</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>{</span>
        <span class='va'>j</span> <span class='op'>&lt;-</span> <span class='va'>celltype.genes</span><span class='op'>[[</span><span class='va'>i</span><span class='op'>]</span><span class='op'>]</span>
        <span class='va'>k</span> <span class='op'>&lt;-</span> <span class='va'>celltype.labels</span> <span class='op'>==</span> <span class='fu'><a href='https://rdrr.io/r/base/paste.html'>paste0</a></span><span class='op'>(</span><span class='st'>"t"</span>, <span class='va'>i</span><span class='op'>)</span>
        <span class='va'>u</span><span class='op'>[</span><span class='va'>j</span>, <span class='va'>k</span><span class='op'>]</span> <span class='op'>&lt;-</span> <span class='fl'>2</span> <span class='op'>*</span> <span class='va'>u</span><span class='op'>[</span><span class='va'>j</span>, <span class='va'>k</span><span class='op'>]</span>
    <span class='op'>}</span>
    <span class='va'>v</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/Log.html'>log2</a></span><span class='op'>(</span><span class='va'>u</span> <span class='op'>+</span> <span class='fl'>1</span><span class='op'>)</span>
    <span class='va'>sce</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/SingleCellExperiment/man/SingleCellExperiment.html'>SingleCellExperiment</a></span><span class='op'>(</span>assays<span class='op'>=</span><span class='fu'><a href='https://rdrr.io/r/base/list.html'>list</a></span><span class='op'>(</span>counts<span class='op'>=</span><span class='va'>u</span>, logcounts<span class='op'>=</span><span class='va'>v</span><span class='op'>)</span><span class='op'>)</span>

    <span class='co'># define marker genes (subset of true cell-type-specific genes)</span>
    <span class='va'>marker.genes</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/BiocGenerics/man/lapply.html'>lapply</a></span><span class='op'>(</span><span class='va'>celltype.genes</span>, <span class='st'>"["</span>, <span class='fl'>1</span><span class='op'>:</span><span class='fl'>5</span><span class='op'>)</span>
    <span class='va'>marker.genes</span>

    <span class='co'># predict cell types</span>
    <span class='va'>res</span> <span class='op'>&lt;-</span> <span class='fu'>labelCells</span><span class='op'>(</span><span class='va'>sce</span>, <span class='va'>marker.genes</span>,
                      fraction_topscoring <span class='op'>=</span> <span class='fl'>0.1</span>,
                      normGenesetExpressionParams <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/list.html'>list</a></span><span class='op'>(</span>R <span class='op'>=</span> <span class='fl'>50</span><span class='op'>)</span><span class='op'>)</span>

    <span class='co'># high-scoring cells used as references for each celltype</span>
    <span class='va'>res</span><span class='op'>$</span><span class='va'>cells</span>

    <span class='co'># ... from these, pseudo-bulks were created:</span>
    <span class='va'>res</span><span class='op'>$</span><span class='va'>refs</span>

    <span class='co'># ... and used to predict labels for all cells</span>
    <span class='va'>res</span><span class='op'>$</span><span class='va'>labels</span><span class='op'>$</span><span class='va'>pruned.labels</span>

    <span class='co'># compare predicted to true cell types</span>
    <span class='fu'><a href='https://rdrr.io/pkg/BiocGenerics/man/table.html'>table</a></span><span class='op'>(</span>true <span class='op'>=</span> <span class='va'>celltype.labels</span>, predicted <span class='op'>=</span> <span class='va'>res</span><span class='op'>$</span><span class='va'>labels</span><span class='op'>$</span><span class='va'>pruned.labels</span><span class='op'>)</span>
<span class='op'>}</span>
</div><div class='co'>#&gt; Loading required package: SummarizedExperiment</div><div class='co'>#&gt; Loading required package: MatrixGenerics</div><div class='co'>#&gt; Loading required package: matrixStats</div><div class='co'>#&gt; 
#&gt; Attaching package: ‘MatrixGenerics’</div><div class='co'>#&gt; The following objects are masked from ‘package:matrixStats’:
#&gt; 
#&gt;     colAlls, colAnyNAs, colAnys, colAvgsPerRowSet, colCollapse,
#&gt;     colCounts, colCummaxs, colCummins, colCumprods, colCumsums,
#&gt;     colDiffs, colIQRDiffs, colIQRs, colLogSumExps, colMadDiffs,
#&gt;     colMads, colMaxs, colMeans2, colMedians, colMins, colOrderStats,
#&gt;     colProds, colQuantiles, colRanges, colRanks, colSdDiffs, colSds,
#&gt;     colSums2, colTabulates, colVarDiffs, colVars, colWeightedMads,
#&gt;     colWeightedMeans, colWeightedMedians, colWeightedSds,
#&gt;     colWeightedVars, rowAlls, rowAnyNAs, rowAnys, rowAvgsPerColSet,
#&gt;     rowCollapse, rowCounts, rowCummaxs, rowCummins, rowCumprods,
#&gt;     rowCumsums, rowDiffs, rowIQRDiffs, rowIQRs, rowLogSumExps,
#&gt;     rowMadDiffs, rowMads, rowMaxs, rowMeans2, rowMedians, rowMins,
#&gt;     rowOrderStats, rowProds, rowQuantiles, rowRanges, rowRanks,
#&gt;     rowSdDiffs, rowSds, rowSums2, rowTabulates, rowVarDiffs, rowVars,
#&gt;     rowWeightedMads, rowWeightedMeans, rowWeightedMedians,
#&gt;     rowWeightedSds, rowWeightedVars</div><div class='co'>#&gt; Loading required package: Biobase</div><div class='co'>#&gt; Welcome to Bioconductor
#&gt; 
#&gt;     Vignettes contain introductory material; view with
#&gt;     'browseVignettes()'. To cite Bioconductor, see
#&gt;     'citation("Biobase")', and for packages 'citation("pkgname")'.</div><div class='co'>#&gt; 
#&gt; Attaching package: ‘Biobase’</div><div class='co'>#&gt; The following object is masked from ‘package:MatrixGenerics’:
#&gt; 
#&gt;     rowMedians</div><div class='co'>#&gt; The following objects are masked from ‘package:matrixStats’:
#&gt; 
#&gt;     anyMissing, rowMedians</div><div class='co'>#&gt;     predicted
#&gt; true t1 t2 t3
#&gt;   t1 30  0  0
#&gt;   t2  0 30  0
#&gt;   t3  0  0 30</div><div class='input'>      
</div></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p><p>Developed by Michael Stadler, <a href="https://csoneson.github.io" class="external-link">Charlotte Soneson</a>, Panagiotis Papasaikas, Dania Machlab.</p></p>
</div>

<div class="pkgdown">
  <p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 1.6.1.9001.</p></p>
</div>

      </footer>
   </div>

  


  

  </body>
</html>


