<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Plot gene region — plotGeneRegion • swissknife</title>


<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->
<link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/cerulean/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous" />


<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>




<meta property="og:title" content="Plot gene region — plotGeneRegion" />
<meta property="og:description" content="Visualize the gene model for a gene of interest, or for all genes in a
provided region, and/or show one or more coverage tracks based on bigwig
file(s)." />






<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">swissknife</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.27</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/fmicompbio/swissknife/">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Plot gene region</h1>
    <small class="dont-index">Source: <a href='https://github.com/fmicompbio/swissknife/blob/master/R/plotGeneRegion.R'><code>R/plotGeneRegion.R</code></a></small>
    <div class="hidden name"><code>plotGeneRegion.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Visualize the gene model for a gene of interest, or for all genes in a
provided region, and/or show one or more coverage tracks based on bigwig
file(s).</p>
    </div>

    <pre class="usage"><span class='fu'>plotGeneRegion</span><span class='op'>(</span>
  gtf <span class='op'>=</span> <span class='st'>""</span>,
  granges <span class='op'>=</span> <span class='cn'>NULL</span>,
  chr <span class='op'>=</span> <span class='st'>""</span>,
  start <span class='op'>=</span> <span class='cn'>NA_real_</span>,
  end <span class='op'>=</span> <span class='cn'>NA_real_</span>,
  showgene <span class='op'>=</span> <span class='st'>""</span>,
  bigwigFiles <span class='op'>=</span> <span class='st'>""</span>,
  bigwigCond <span class='op'>=</span> <span class='st'>""</span>,
  geneTrackTitle <span class='op'>=</span> <span class='st'>"Genes"</span>,
  transcriptIdColumn <span class='op'>=</span> <span class='st'>"transcript_id"</span>,
  geneIdColumn <span class='op'>=</span> <span class='st'>"gene_id"</span>,
  geneSymbolColumn <span class='op'>=</span> <span class='st'>"gene_name"</span>,
  lowerPadding <span class='op'>=</span> <span class='fl'>0.15</span>,
  upperPadding <span class='op'>=</span> <span class='fl'>0.05</span>,
  colorByStrand <span class='op'>=</span> <span class='cn'>FALSE</span>,
  featureColors <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span>plusmain <span class='op'>=</span> <span class='st'>"#0E14D0"</span>, minusmain <span class='op'>=</span> <span class='st'>"#D0350E"</span>, plusother <span class='op'>=</span> <span class='st'>"#9E9BEB"</span>,
    minusother <span class='op'>=</span> <span class='st'>"#DA907E"</span><span class='op'>)</span>,
  condColors <span class='op'>=</span> <span class='cn'>NULL</span>,
  scaleDataTracks <span class='op'>=</span> <span class='cn'>FALSE</span>
<span class='op'>)</span></pre>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>gtf</th>
      <td><p>Character scalar, path to gtf file (tested with Ensembl/Gencode
files).</p></td>
    </tr>
    <tr>
      <th>granges</th>
      <td><p>GRanges object, typically generated from a GTF file using the
<code>prepareGTF</code> function. This is an alternative to providing the link 
to the gtf file directly, and will take precedence over the <code>gtf</code> 
argument if provided.</p></td>
    </tr>
    <tr>
      <th>chr</th>
      <td><p>Character scalar, name of the chromosome to show.</p></td>
    </tr>
    <tr>
      <th>start, end</th>
      <td><p>Numeric scalars, start and end position of the region to
show.</p></td>
    </tr>
    <tr>
      <th>showgene</th>
      <td><p>Character scalar, the gene ID/name to display. Will take 
precedence over positional range specification if provided.</p></td>
    </tr>
    <tr>
      <th>bigwigFiles</th>
      <td><p>Named character vector, paths to bigwig files.</p></td>
    </tr>
    <tr>
      <th>bigwigCond</th>
      <td><p>Named character vector, the grouping of the bigwig files
(used for coloring of the coverage tracks).</p></td>
    </tr>
    <tr>
      <th>geneTrackTitle</th>
      <td><p>Character scalar, name of the gene track.</p></td>
    </tr>
    <tr>
      <th>transcriptIdColumn</th>
      <td><p>Character scalar, the column in the gtf file that 
contains the transcript ID. Passed to <code>prepareGTF</code>.</p></td>
    </tr>
    <tr>
      <th>geneIdColumn</th>
      <td><p>Character scalar, the column in the gtf file that 
contains the gene ID. Passed to <code>prepareGTF</code>.</p></td>
    </tr>
    <tr>
      <th>geneSymbolColumn</th>
      <td><p>Character scalar, the column in the gtf file that 
contains the gene symbol (if available). Set to <code>""</code> if not 
available (in which case the gene IDs will be used in its place).
Passed to <code>prepareGTF</code>.</p></td>
    </tr>
    <tr>
      <th>lowerPadding, upperPadding</th>
      <td><p>Numeric scalars, setting the amount of 
padding in the lower and upper range of the plot, respectively. For 
example, a value of 0.05 will expand the range by 
0.05 * (max coordinate - min coordinate) in the specified direction.</p></td>
    </tr>
    <tr>
      <th>colorByStrand</th>
      <td><p>Logical scalar, determining whether gene features are 
colored by the annotated strand.</p></td>
    </tr>
    <tr>
      <th>featureColors</th>
      <td><p>Named character vector of length 4, with elements 
<code>plusmain</code>, <code>minusmain</code>, <code>plusother</code>, <code>minusother</code>, 
giving the colors to use for the features if <code>colorByStrand</code> is TRUE.</p></td>
    </tr>
    <tr>
      <th>condColors</th>
      <td><p>Either NULL or a named character vector (with the same 
names as the unique values of <code>bigwigCond</code>), giving the colors to use
for the coverage tracks if <code>bigwigCond</code> is provided.</p></td>
    </tr>
    <tr>
      <th>scaleDataTracks</th>
      <td><p>Logical scalar, indicating whether the data tracks 
should be scaled to have the same y-axis limits.</p></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="details"><a class="anchor" href="#details"></a>Details</h2>

    <p>The gene annotation can be provided either as a path to a gtf file, or as a 
GRanges object (generated using the <code>prepareGTF</code> function to ensure 
compatibility). The region to display can be determined either by 
specifying a gene (ID or symbol) or by specifying a viewing range 
(chromosome, start and end positions).</p>
    <h2 class="hasAnchor" id="author"><a class="anchor" href="#author"></a>Author</h2>

    <p>Charlotte Soneson</p>

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'><span class='va'>gtffile</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/system.file.html'>system.file</a></span><span class='op'>(</span><span class='st'>"extdata/plotGeneRegion/mm10_ensembl98.gtf"</span>, 
                       package <span class='op'>=</span> <span class='st'>"swissknife"</span><span class='op'>)</span>
<span class='fu'>plotGeneRegion</span><span class='op'>(</span>gtf <span class='op'>=</span> <span class='va'>gtffile</span>, 
               showgene <span class='op'>=</span> <span class='st'>"Tnfaip3"</span><span class='op'>)</span>
</div><div class='img'><img src='plotGeneRegion-1.png' alt='' width='700' height='433' /></div><div class='input'>               
<span class='va'>bwf</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/system.file.html'>system.file</a></span><span class='op'>(</span><span class='st'>"extdata/plotGeneRegion/mnase_mm10.bw"</span>, 
                   package <span class='op'>=</span> <span class='st'>"swissknife"</span><span class='op'>)</span>
<span class='fu'><a href='https://rdrr.io/r/base/names.html'>names</a></span><span class='op'>(</span><span class='va'>bwf</span><span class='op'>)</span> <span class='op'>&lt;-</span> <span class='st'>"bwf1"</span>
<span class='fu'>plotGeneRegion</span><span class='op'>(</span>gtf <span class='op'>=</span> <span class='va'>gtffile</span>, 
               bigwigFiles <span class='op'>=</span> <span class='va'>bwf</span>,
               chr <span class='op'>=</span> <span class='st'>"chr10"</span>, start <span class='op'>=</span> <span class='fl'>20000000</span>, end <span class='op'>=</span> <span class='fl'>20005000</span><span class='op'>)</span>
</div><div class='img'><img src='plotGeneRegion-2.png' alt='' width='700' height='433' /></div><div class='input'><span class='fu'>plotGeneRegion</span><span class='op'>(</span>bigwigFiles <span class='op'>=</span> <span class='va'>bwf</span>,
               chr <span class='op'>=</span> <span class='st'>"chr10"</span>, start <span class='op'>=</span> <span class='fl'>20000000</span>, end <span class='op'>=</span> <span class='fl'>20005000</span><span class='op'>)</span>
</div><div class='img'><img src='plotGeneRegion-3.png' alt='' width='700' height='433' /></div><div class='input'>               
<span class='va'>bwf2</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='va'>bwf</span>, <span class='va'>bwf</span><span class='op'>)</span>
<span class='fu'><a href='https://rdrr.io/r/base/names.html'>names</a></span><span class='op'>(</span><span class='va'>bwf2</span><span class='op'>)</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"bwf1"</span>, <span class='st'>"bwf2"</span><span class='op'>)</span>
<span class='va'>bwc2</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"c1"</span>, <span class='st'>"c2"</span><span class='op'>)</span>
<span class='fu'><a href='https://rdrr.io/r/base/names.html'>names</a></span><span class='op'>(</span><span class='va'>bwc2</span><span class='op'>)</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/names.html'>names</a></span><span class='op'>(</span><span class='va'>bwf2</span><span class='op'>)</span>
<span class='fu'>plotGeneRegion</span><span class='op'>(</span>gtf <span class='op'>=</span> <span class='va'>gtffile</span>, bigwigFiles <span class='op'>=</span> <span class='va'>bwf2</span>, bigwigCond <span class='op'>=</span> <span class='va'>bwc2</span>, 
               showgene <span class='op'>=</span> <span class='st'>"Map3k5"</span><span class='op'>)</span>
</div><div class='img'><img src='plotGeneRegion-4.png' alt='' width='700' height='433' /></div><div class='input'>
</div></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by Michael Stadler, <a href='https://csoneson.github.io'>Charlotte Soneson</a>, Panagiotis Papasaikas, Dania Machlab.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.9000.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


