<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Plot gene region — plotGeneRegion • swissknife</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/cerulean/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Plot gene region — plotGeneRegion"><meta property="og:description" content="Visualize the gene model for a gene of interest, or for all genes in a
provided region, and/or show one or more coverage tracks based on bigwig
file(s)."><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">swissknife</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.36</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/fmicompbio/swissknife/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Plot gene region</h1>
    <small class="dont-index">Source: <a href="https://github.com/fmicompbio/swissknife/blob/HEAD/R/plotGeneRegion.R" class="external-link"><code>R/plotGeneRegion.R</code></a></small>
    <div class="hidden name"><code>plotGeneRegion.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Visualize the gene model for a gene of interest, or for all genes in a
provided region, and/or show one or more coverage tracks based on bigwig
file(s).</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="fu">plotGeneRegion</span><span class="op">(</span>
  gtf <span class="op">=</span> <span class="st">""</span>,
  granges <span class="op">=</span> <span class="cn">NULL</span>,
  chr <span class="op">=</span> <span class="st">""</span>,
  start <span class="op">=</span> <span class="cn">NA_real_</span>,
  end <span class="op">=</span> <span class="cn">NA_real_</span>,
  showgene <span class="op">=</span> <span class="st">""</span>,
  bigwigFiles <span class="op">=</span> <span class="st">""</span>,
  bigwigCond <span class="op">=</span> <span class="st">""</span>,
  geneTrackTitle <span class="op">=</span> <span class="st">"Genes"</span>,
  transcriptIdColumn <span class="op">=</span> <span class="st">"transcript_id"</span>,
  geneIdColumn <span class="op">=</span> <span class="st">"gene_id"</span>,
  geneSymbolColumn <span class="op">=</span> <span class="st">"gene_name"</span>,
  lowerPadding <span class="op">=</span> <span class="fl">0.15</span>,
  upperPadding <span class="op">=</span> <span class="fl">0.05</span>,
  colorByStrand <span class="op">=</span> <span class="cn">FALSE</span>,
  featureColors <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">c</a></span><span class="op">(</span>plusmain <span class="op">=</span> <span class="st">"#0E14D0"</span>, minusmain <span class="op">=</span> <span class="st">"#D0350E"</span>, plusother <span class="op">=</span> <span class="st">"#9E9BEB"</span>,
    minusother <span class="op">=</span> <span class="st">"#DA907E"</span><span class="op">)</span>,
  condColors <span class="op">=</span> <span class="cn">NULL</span>,
  scaleDataTracks <span class="op">=</span> <span class="cn">FALSE</span>
<span class="op">)</span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>
    <dl><dt>gtf</dt>
<dd><p>Character scalar, path to gtf file (tested with Ensembl/Gencode
files).</p></dd>
<dt>granges</dt>
<dd><p>GRanges object, typically generated from a GTF file using the
<code>prepareGTF</code> function. This is an alternative to providing the link 
to the gtf file directly, and will take precedence over the <code>gtf</code> 
argument if provided.</p></dd>
<dt>chr</dt>
<dd><p>Character scalar, name of the chromosome to show.</p></dd>
<dt>start, end</dt>
<dd><p>Numeric scalars, start and end position of the region to
show.</p></dd>
<dt>showgene</dt>
<dd><p>Character scalar, the gene ID/name to display. Will take 
precedence over positional range specification if provided.</p></dd>
<dt>bigwigFiles</dt>
<dd><p>Named character vector, paths to bigwig files.</p></dd>
<dt>bigwigCond</dt>
<dd><p>Named character vector, the grouping of the bigwig files
(used for coloring of the coverage tracks).</p></dd>
<dt>geneTrackTitle</dt>
<dd><p>Character scalar, name of the gene track.</p></dd>
<dt>transcriptIdColumn</dt>
<dd><p>Character scalar, the column in the gtf file that 
contains the transcript ID. Passed to <code>prepareGTF</code>.</p></dd>
<dt>geneIdColumn</dt>
<dd><p>Character scalar, the column in the gtf file that 
contains the gene ID. Passed to <code>prepareGTF</code>.</p></dd>
<dt>geneSymbolColumn</dt>
<dd><p>Character scalar, the column in the gtf file that 
contains the gene symbol (if available). Set to <code>""</code> if not 
available (in which case the gene IDs will be used in its place).
Passed to <code>prepareGTF</code>.</p></dd>
<dt>lowerPadding, upperPadding</dt>
<dd><p>Numeric scalars, setting the amount of 
padding in the lower and upper range of the plot, respectively. For 
example, a value of 0.05 will expand the range by 
0.05 * (max coordinate - min coordinate) in the specified direction.</p></dd>
<dt>colorByStrand</dt>
<dd><p>Logical scalar, determining whether gene features are 
colored by the annotated strand.</p></dd>
<dt>featureColors</dt>
<dd><p>Named character vector of length 4, with elements 
<code>plusmain</code>, <code>minusmain</code>, <code>plusother</code>, <code>minusother</code>, 
giving the colors to use for the features if <code>colorByStrand</code> is TRUE.</p></dd>
<dt>condColors</dt>
<dd><p>Either NULL or a named character vector (with the same 
names as the unique values of <code>bigwigCond</code>), giving the colors to use
for the coverage tracks if <code>bigwigCond</code> is provided.</p></dd>
<dt>scaleDataTracks</dt>
<dd><p>Logical scalar, indicating whether the data tracks 
should be scaled to have the same y-axis limits.</p></dd>
</dl></div>
    <div id="details">
    <h2>Details</h2>
    <p>The gene annotation can be provided either as a path to a gtf file, or as a 
GRanges object (generated using the <code>prepareGTF</code> function to ensure 
compatibility). The region to display can be determined either by 
specifying a gene (ID or symbol) or by specifying a viewing range 
(chromosome, start and end positions).</p>
    </div>
    <div id="author">
    <h2>Author</h2>
    <p>Charlotte Soneson</p>
    </div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span class="kw">if</span> <span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html" class="external-link">requireNamespace</a></span><span class="op">(</span><span class="st">"Gviz"</span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span class="r-in">    <span class="va">gtffile</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata/plotGeneRegion/mm10_ensembl98.gtf"</span>, </span>
<span class="r-in">                           package <span class="op">=</span> <span class="st">"swissknife"</span><span class="op">)</span></span>
<span class="r-in">    <span class="fu">plotGeneRegion</span><span class="op">(</span>gtf <span class="op">=</span> <span class="va">gtffile</span>, </span>
<span class="r-in">                   showgene <span class="op">=</span> <span class="st">"Tnfaip3"</span><span class="op">)</span></span>
<span class="r-in">               </span>
<span class="r-in">    <span class="va">bwf</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata/plotGeneRegion/mnase_mm10.bw"</span>, </span>
<span class="r-in">                       package <span class="op">=</span> <span class="st">"swissknife"</span><span class="op">)</span></span>
<span class="r-in">    <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">bwf</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="st">"bwf1"</span></span>
<span class="r-in">    <span class="fu">plotGeneRegion</span><span class="op">(</span>gtf <span class="op">=</span> <span class="va">gtffile</span>, </span>
<span class="r-in">                   bigwigFiles <span class="op">=</span> <span class="va">bwf</span>,</span>
<span class="r-in">                   chr <span class="op">=</span> <span class="st">"chr10"</span>, start <span class="op">=</span> <span class="fl">20000000</span>, end <span class="op">=</span> <span class="fl">20005000</span><span class="op">)</span></span>
<span class="r-in">    <span class="fu">plotGeneRegion</span><span class="op">(</span>bigwigFiles <span class="op">=</span> <span class="va">bwf</span>,</span>
<span class="r-in">                   chr <span class="op">=</span> <span class="st">"chr10"</span>, start <span class="op">=</span> <span class="fl">20000000</span>, end <span class="op">=</span> <span class="fl">20005000</span><span class="op">)</span></span>
<span class="r-in">               </span>
<span class="r-in">    <span class="va">bwf2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">c</a></span><span class="op">(</span><span class="va">bwf</span>, <span class="va">bwf</span><span class="op">)</span></span>
<span class="r-in">    <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">bwf2</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"bwf1"</span>, <span class="st">"bwf2"</span><span class="op">)</span></span>
<span class="r-in">    <span class="va">bwc2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"c1"</span>, <span class="st">"c2"</span><span class="op">)</span></span>
<span class="r-in">    <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">bwc2</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">bwf2</span><span class="op">)</span></span>
<span class="r-in">    <span class="fu">plotGeneRegion</span><span class="op">(</span>gtf <span class="op">=</span> <span class="va">gtffile</span>, bigwigFiles <span class="op">=</span> <span class="va">bwf2</span>, bigwigCond <span class="op">=</span> <span class="va">bwc2</span>, </span>
<span class="r-in">                   showgene <span class="op">=</span> <span class="st">"Map3k5"</span><span class="op">)</span></span>
<span class="r-in"><span class="op">}</span></span>
<span class="r-plt img"><img src="plotGeneRegion-1.png" alt="" width="700" height="433"></span>
<span class="r-plt img"><img src="plotGeneRegion-2.png" alt="" width="700" height="433"></span>
<span class="r-plt img"><img src="plotGeneRegion-3.png" alt="" width="700" height="433"></span>
<span class="r-plt img"><img src="plotGeneRegion-4.png" alt="" width="700" height="433"></span>
<span class="r-in"></span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Michael Stadler, <a href="https://csoneson.github.io" class="external-link">Charlotte Soneson</a>, Panagiotis Papasaikas, Dania Machlab.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.1.9000.</p>
</div>

      </footer></div>

  


  

  </body></html>

