<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Select Variable Genes in Single Cell RNA-seq — selVarGenes • swissknife</title>


<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->
<link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/cerulean/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous" />


<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>




<meta property="og:title" content="Select Variable Genes in Single Cell RNA-seq — selVarGenes" />
<meta property="og:description" content="This function selects the most variable genes from a SingleCellExperiment object
  using the plot that displays the log2 coefficient of variation as a function of the log2 mean for 
  all genes across all the cells." />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">swissknife</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.21</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/fmicompbio/swissknife/">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Select Variable Genes in Single Cell RNA-seq</h1>
    <small class="dont-index">Source: <a href='https://github.com/fmicompbio/swissknife/blob/master/R/selVarGenes.R'><code>R/selVarGenes.R</code></a></small>
    <div class="hidden name"><code>selVarGenes.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>This function selects the most variable genes from a <code>SingleCellExperiment</code> object
  using the plot that displays the log2 coefficient of variation as a function of the log2 mean for 
  all genes across all the cells.</p>
    </div>

    <pre class="usage"><span class='fu'>selVarGenes</span><span class='op'>(</span>
  data <span class='op'>=</span> <span class='cn'>NULL</span>,
  assay.type <span class='op'>=</span> <span class='st'>"counts"</span>,
  logPseudo <span class='op'>=</span> <span class='fl'>1</span>,
  Nmads <span class='op'>=</span> <span class='fl'>3</span>,
  minCells <span class='op'>=</span> <span class='fl'>5</span>,
  minExpr <span class='op'>=</span> <span class='fl'>1</span>,
  exclTopExprFrac <span class='op'>=</span> <span class='fl'>0.01</span>,
  span <span class='op'>=</span> <span class='fl'>0.2</span>,
  control <span class='op'>=</span> <span class='fu'>stats</span><span class='fu'>::</span><span class='fu'><a href='https://rdrr.io/r/stats/loess.control.html'>loess.control</a></span><span class='op'>(</span>surface <span class='op'>=</span> <span class='st'>"direct"</span><span class='op'>)</span>,
  nBins <span class='op'>=</span> <span class='fl'>100</span>,
  nBinsDense <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/Round.html'>ceiling</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/nrow.html'>nrow</a></span><span class='op'>(</span><span class='va'>data</span><span class='op'>)</span><span class='op'>/</span><span class='fl'>4</span><span class='op'>)</span>,
  <span class='va'>...</span>
<span class='op'>)</span></pre>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>data</th>
      <td><p><code>SingleCellExperiment</code> object or normalized count <code>matrix</code> containing the genes as rows and cells as columns.</p></td>
    </tr>
    <tr>
      <th>assay.type</th>
      <td><p>the type of assay to use if <code>data</code> is a <code>SingleCellExperiment</code>. It can be either 'counts' 
or 'logcounts'. The default is 'counts'.</p></td>
    </tr>
    <tr>
      <th>logPseudo</th>
      <td><p>pseudo-count to use when using the logcounts slot from the <code>SingleCellExperiment</code> to transform back to
normalized raw count space.</p></td>
    </tr>
    <tr>
      <th>Nmads</th>
      <td><p>number of MADs beyond which genes are selected per bin.</p></td>
    </tr>
    <tr>
      <th>minCells</th>
      <td><p>keep genes with minimum expression in at least this number of cells.</p></td>
    </tr>
    <tr>
      <th>minExpr</th>
      <td><p>keep genes with expression greater than or equal to this in <code>minCells</code> cells in the 
normalized count matrix.</p></td>
    </tr>
    <tr>
      <th>exclTopExprFrac</th>
      <td><p>the fraction of top expressed genes that will be excluded from the loess fit 
(value between 0 and 1).</p></td>
    </tr>
    <tr>
      <th>span</th>
      <td><p>span parameter for <code>loess</code> function.</p></td>
    </tr>
    <tr>
      <th>control</th>
      <td><p>control parameters for <code>loess</code> function.</p></td>
    </tr>
    <tr>
      <th>nBins</th>
      <td><p>number of bins or groups to place the points(genes) into.</p></td>
    </tr>
    <tr>
      <th>nBinsDense</th>
      <td><p>number of bins or groups to use  to place the points(genes) into
when calculating more accurate distance values to the curve from the loess fit.</p></td>
    </tr>
    <tr>
      <th>...</th>
      <td><p>additional parameters for the <code>loess</code> function from the <code>stats</code> package.</p></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>a <code>list</code> of length 2:</p><ul>
<li><p>varGenes: <code>vector</code> containing the names of the most variable genes.</p></li>
<li><p>geneInfo: <code>data.frame</code> with genes as rows and columns containing  
                     calculated measures:</p><ul>
<li><p>logMean: log2(mean) expression of genes across cells.</p></li>
<li><p>logCV: log2(coefficient of variation) of genes across cells.</p></li>
<li><p>pred_logCV: predicted log2(coefficient of variation) from loess fit.</p></li>
<li><p>assigned_bin: bin each gene has been assigned to.</p></li>
<li><p>distance: accurate distance measuses. Points below the loess fit get a negative sign.</p></li>
</ul></li>
</ul>

    <h2 class="hasAnchor" id="details"><a class="anchor" href="#details"></a>Details</h2>

    <p>The function takes in a <code>SingleCellExperiment</code> object and calculates the normalized
  counts by dividing the raw counts by the corresponding sizeFactors per cell, or a matrix of already normalized counts. 
  Only genes that have an expression greater than or equal to <code>minExpr</code> in at least <code>minCells</code> cells will be
  kept. If <code>assay.type</code> is set to 'logcounts', that assay is transformed back to the raw normalized count space
  by performing 2^logcounts(data) - 1, under the assumption the logcounts data is in log2 form and had a pseudocount of 1.</p>  
<p>The genes that vary most on the log2(coefficient of variation) vs log2(mean) plot of genes will 
  be selected. A loess fit is done on this plot and the distance (euclidean by default) each point has
  to the curve is calculated in two steps.</p>  
<p>In the first step, genes are assigned to bins by taking the minimum distance to the curve. By default
  we select 100 points on the loess fit and calculate the distances each gene has to all those points on 
  the curve. Each gene is assigned to the point on the curve for which it has the shortest distance. In 
  the second step, more accurate distances to the curve are calculated by using a higher number of 
  points on the curve. Distances are calculated using the <code><a href='https://rdrr.io/pkg/wordspace/man/dist_matrix.html'>dist.matrix</a></code> function.</p>  
<p>Finally, for each bin, the most variable genes are selected using the more accurate distance
  measures. Genes that fall below the loess fit are assigned a negative sign and the genes that 
  are <code>Nmads</code> MADs away from the median are selected.</p>
    <h2 class="hasAnchor" id="author"><a class="anchor" href="#author"></a>Author</h2>

    <p>Dania Machlab</p>

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'>   <span class='co'># packages</span>
   <span class='kw'><a href='https://rdrr.io/r/base/library.html'>library</a></span><span class='op'>(</span><span class='va'>SingleCellExperiment</span><span class='op'>)</span>
   
   <span class='co'># create example count matrix</span>
   <span class='co'># ... poisson distr per gene</span>
   <span class='va'>mu</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/Round.html'>ceiling</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/stats/Uniform.html'>runif</a></span><span class='op'>(</span>n <span class='op'>=</span> <span class='fl'>2000</span>, min <span class='op'>=</span> <span class='fl'>0</span>, max <span class='op'>=</span> <span class='fl'>100</span><span class='op'>)</span><span class='op'>)</span>
   <span class='va'>counts</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/do.call.html'>do.call</a></span><span class='op'>(</span><span class='va'>rbind</span>, <span class='fu'><a href='https://rdrr.io/r/base/lapply.html'>lapply</a></span><span class='op'>(</span><span class='va'>mu</span>, <span class='kw'>function</span><span class='op'>(</span><span class='va'>x</span><span class='op'>)</span><span class='op'>{</span><span class='fu'><a href='https://rdrr.io/r/stats/Poisson.html'>rpois</a></span><span class='op'>(</span><span class='fl'>1000</span>, lambda <span class='op'>=</span> <span class='va'>x</span><span class='op'>)</span><span class='op'>}</span><span class='op'>)</span><span class='op'>)</span>
   <span class='va'>counts</span> <span class='op'>&lt;-</span> <span class='va'>counts</span> <span class='op'>+</span> <span class='fl'>1</span>
   <span class='co'># ... add signal to subset of genes (rows) and cells (columns)</span>
   <span class='va'>i</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/sample.html'>sample</a></span><span class='op'>(</span>x <span class='op'>=</span> <span class='fl'>1</span><span class='op'>:</span><span class='fu'><a href='https://rdrr.io/r/base/nrow.html'>nrow</a></span><span class='op'>(</span><span class='va'>counts</span><span class='op'>)</span>, size <span class='op'>=</span> <span class='fl'>500</span><span class='op'>)</span>
   <span class='va'>j</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/sample.html'>sample</a></span><span class='op'>(</span>x <span class='op'>=</span> <span class='fl'>1</span><span class='op'>:</span><span class='fu'><a href='https://rdrr.io/r/base/nrow.html'>ncol</a></span><span class='op'>(</span><span class='va'>counts</span><span class='op'>)</span>, size <span class='op'>=</span> <span class='fl'>500</span><span class='op'>)</span>
   <span class='va'>counts</span><span class='op'>[</span><span class='va'>i</span>, <span class='va'>j</span><span class='op'>]</span> <span class='op'>&lt;-</span> <span class='va'>counts</span><span class='op'>[</span><span class='va'>i</span>, <span class='va'>j</span><span class='op'>]</span> <span class='op'>+</span> <span class='fu'><a href='https://rdrr.io/r/base/sample.html'>sample</a></span><span class='op'>(</span><span class='fl'>5</span><span class='op'>:</span><span class='fl'>10</span>, <span class='fu'><a href='https://rdrr.io/r/base/length.html'>length</a></span><span class='op'>(</span><span class='va'>i</span><span class='op'>)</span>, replace <span class='op'>=</span> <span class='cn'>TRUE</span><span class='op'>)</span>
   
   <span class='co'># create SCE </span>
   <span class='va'>sce</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/SingleCellExperiment/man/SingleCellExperiment.html'>SingleCellExperiment</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/list.html'>list</a></span><span class='op'>(</span>counts<span class='op'>=</span><span class='va'>counts</span><span class='op'>)</span><span class='op'>)</span>
   
   <span class='co'># calculate sizeFactors</span>
   <span class='va'>libsizes</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/colSums.html'>colSums</a></span><span class='op'>(</span><span class='va'>counts</span><span class='op'>)</span>
   <span class='fu'><a href='https://rdrr.io/pkg/SingleCellExperiment/man/sizeFactors.html'>sizeFactors</a></span><span class='op'>(</span><span class='va'>sce</span><span class='op'>)</span> <span class='op'>&lt;-</span> <span class='va'>libsizes</span><span class='op'>/</span><span class='fu'><a href='https://rdrr.io/r/base/mean.html'>mean</a></span><span class='op'>(</span><span class='va'>libsizes</span><span class='op'>)</span>
 
   <span class='co'># select variable genes</span>
   <span class='va'>varGenes</span> <span class='op'>&lt;-</span> <span class='fu'>selVarGenes</span><span class='op'>(</span><span class='va'>sce</span>, assay.type<span class='op'>=</span><span class='st'>"counts"</span><span class='op'>)</span>
</div><div class='output co'>#&gt; <span class='message'>Count row names are empty, naming them now ... </span></div><div class='output co'>#&gt; <span class='message'>Count column names are empty, naming them now ... </span></div><div class='input'>   
   <span class='co'># plot</span>
   <span class='fu'><a href='plotSelVarGenes.html'>plotSelVarGenes</a></span><span class='op'>(</span><span class='va'>varGenes</span>, colByBin<span class='op'>=</span><span class='cn'>TRUE</span><span class='op'>)</span>
</div><div class='img'><img src='selVarGenes-1.png' alt='' width='700' height='433' /></div><div class='input'>   <span class='fu'><a href='plotSelVarGenes.html'>plotSelVarGenes</a></span><span class='op'>(</span><span class='va'>varGenes</span><span class='op'>)</span>
</div><div class='img'><img src='selVarGenes-2.png' alt='' width='700' height='433' /></div><div class='input'>
</div></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by Michael Stadler, <a href='https://csoneson.github.io'>Charlotte Soneson</a>, Panagiotis Papasaikas, Dania Machlab.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.9000.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


